### PR TITLE
Fix the compilation warnings

### PR DESCRIPTION
**Pull request by liqinhuixm**
## Summary
Fix the compilation warnings
## Impact
N/A
## Testing
N/A

### CODE REVIEW

**Review by jerpelea — APPROVED** [review]

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_netdev.c [hunk 1]
```diff
@@ -865,8 +865,6 @@ static int bcmf_txavail(FAR struct net_driver_s *dev)
 #if defined(CONFIG_NET_MCASTGROUP) || defined(CONFIG_NET_ICMPv6)
 static int bcmf_addmac(FAR struct net_driver_s *dev, FAR const uint8_t *mac)
 {
-  FAR struct bcmf_dev_s *priv = (FAR struct bcmf_dev_s *)dev->d_private;
-
   /* Add the MAC address to the hardware multicast routing table */
 
   UNUSED(priv);
```
Comment (anchao) [inline drivers/wireless/ieee80211/bcm43xxx/bcmf_netdev.c:870]:
```suggestion
```

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_netdev.c [hunk 2]
```diff
@@ -895,8 +893,6 @@ static int bcmf_addmac(FAR struct net_driver_s *dev, FAR const uint8_t *mac)
 #ifdef CONFIG_NET_MCASTGROUP
 static int bcmf_rmmac(FAR struct net_driver_s *dev, FAR const uint8_t *mac)
 {
-  FAR struct bcmf_dev_s *priv = (FAR struct bcmf_dev_s *)dev->d_private;
-
   /* Add the MAC address to the hardware multicast routing table */
 
   UNUSED(priv);
```
Comment (anchao) [inline drivers/wireless/ieee80211/bcm43xxx/bcmf_netdev.c:898]:
```suggestion
```

**Review by anchao — CHANGES_REQUESTED** [review]

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_netdev.c (hunk 1 above)
Comment (liqinhuixm) [reply drivers/wireless/ieee80211/bcm43xxx/bcmf_netdev.c]: Done

File: drivers/wireless/ieee80211/bcm43xxx/bcmf_netdev.c (hunk 2 above)
Comment (liqinhuixm) [reply drivers/wireless/ieee80211/bcm43xxx/bcmf_netdev.c]: Done

**Review by anchao — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**